### PR TITLE
Changing the ssh key gen algorithm for FIPS machines

### PR DESCRIPTION
**Pull request by junaid18183**
Changing the ssh key gen algorithm for FIPS machines 
Fixes the - https://github.com/weaveworks/ignite/issues/135

### CODE REVIEW

File: pkg/metadata/vmmd/fips.go [hunk 1]
```diff
@@ -0,0 +1,18 @@
+package vmmd
+
+import (
+	"os"
+)
+
+// FIPSEnabled returns true if running in FIPS mode.
+// currently it just checks the system wide /etc/system-fips file present or not.
+// We can improve it later.
+func FIPSEnabled() bool {
```
Comment (luxas) [inline pkg/metadata/vmmd/fips.go]: Please move this to `pkg/util`, that seems like a better place for it.

File: pkg/metadata/vmmd/vm.go [hunk 2]
```diff
@@ -193,9 +193,13 @@ func (md *VM) ClearIPAddresses() {
 // Generate a new SSH keypair for the vm
 func (md *VM) newSSHKeypair() (string, error) {
 	privKeyPath := path.Join(md.ObjectPath(), fmt.Sprintf(constants.VM_SSH_KEY_TEMPLATE, md.GetUID()))
-
 	// Use ED25519 instead of RSA for performance (it's equally secure, but a lot faster to generate/authenticate)
-	_, err := util.ExecuteCommand("ssh-keygen", "-q", "-t", "ed25519", "-N", "", "-f", privKeyPath)
+	ssh_key_algorithm := "ed25519"
```
Comment (luxas) [inline pkg/metadata/vmmd/vm.go]: Please add a note here like this:

```go
// TODO: In future versions, let the user specify what key algorithm to use through the API types
```

File: pkg/metadata/vmmd/fips.go (hunk 1 above)
Comment (junaid18183) [reply pkg/metadata/vmmd/fips.go]: Fixed

File: pkg/metadata/vmmd/vm.go (hunk 2 above)
Comment (junaid18183) [reply pkg/metadata/vmmd/vm.go]: Fixed

**Comment by junaid18183** [comment]
Thanks for the review. Have implemented the changes.

**Pull request merged**